patches/minecraft/net/minecraft/item/Item.java.patch

```diff
@@ -337,18 +337,34 @@
 +
 +    /**
 +     * Determines if the specific ItemStack can be placed in the specified armor slot.
++     * @deprecated Override {@link #getEquipmentSlot(ItemStack)} to set the armor slot, and use
++     * {@code armorType == EntityLiving.getSlotForItemStack(stack)} to check the armor slot.
 +     *
 +     * @param stack The ItemStack
 +     * @param armorType Armor slot ID: 0: Helmet, 1: Chest, 2: Legs, 3: Boots
 +     * @param entity The entity trying to equip the armor
 +     * @return True if the given ItemStack can be inserted in the slot
 +     */
++    @Deprecated
 +    public boolean isValidArmor(ItemStack stack, EntityEquipmentSlot armorType, Entity entity)
 +    {
 +        return net.minecraft.entity.EntityLiving.func_184640_d(stack) == armorType;
 +    }
 +
 +    /**
++     * <em><strong>MODDERS</strong>: Override this to set a non-default {@link EntityEquipmentSlot}
```

Comment:
Can you reword this to be less strongly worded? Maybe

```
Override this method to return {@code EntityEquipmentSlot} the given {@code ItemStack} should go in, but <em>do not directly call this to get the {@code EntityEquipmentSlot}</em>. Instead, use {@link ...}.
```

Also, do not link so aggressively. A good rule is to link the first occurrence of a term and then to simply `{@code}` the others. Additionally, try to remove ones that should be known from context even then.